### PR TITLE
FEATURE: Add support for Mistral models

### PR DESCRIPTION
**Pull request by SamSaffron**
Adds support for mistral models (pixtral and mistral large now have presets) 

Also corrects token accounting in AWS bedrock models

### CODE REVIEW

File: lib/completions/endpoints/mistral.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module DiscourseAi
+  module Completions
+    module Endpoints
+      class Mistral < OpenAi
```
Comment (tyb-talks) [inline lib/completions/endpoints/mistral.rb:6]: Should the `provider_id` be overridden? As-is, this currently returns the one for OpenAi.

**Review by tyb-talks — APPROVED** [review]
Minor question on provider_id for the endpoint, but otherwise looks ok.

File: lib/completions/endpoints/mistral.rb (hunk 1 above)
Comment (SamSaffron) [reply lib/completions/endpoints/mistral.rb:6]: oh sure ... good catch will fix

**Review by tyb-talks — CHANGES_REQUESTED** [review]
I just spotted something I missed earlier in the OpenAi endpoint:

```
        def prepare_payload(prompt, model_params, dialect)
          payload = default_options.merge(model_params).merge(messages: prompt)

          if @streaming_mode
            payload[:stream] = true

            # Usage is not available in Azure yet.
            # We'll fallback to guess this using the tokenizer.
            payload[:stream_options] = { include_usage: true } if llm_model.provider == "open_ai"
          end
```

**Review by tyb-talks — APPROVED** [review]
😅 actually it's fine, it just omits those params if it's not from OpenAi. 🤔 If the logic in the inherited endpoint changes more specifically for OpenAi though, it would also affect this Mistral endpoint. Still fine for now.

**Comment by SamSaffron** [comment]
Yeah this is safe by "side effect" cause mistral just sends this info anyway.

**Comment by SamSaffron** [comment]
thanks for the review @tyb-talks :hugs:

**Pull request merged**